server/client.go

```diff
@@ -1588,6 +1749,29 @@ func (c *client) processMsg(msg []byte) {
 		}
 	}
 
+	// Check to see if we need to route this message to
+	// another account.
+	if c.typ == CLIENT && c.acc != nil && c.acc.imports.services != nil {
+		c.acc.mu.RLock()
+		rm := c.acc.imports.services[string(c.pa.subject)]
+		c.acc.mu.RUnlock()
+		// Get the results from the other account for the mapped "to" subject.
+		if rm != nil && rm.acc != nil && rm.acc.sl != nil {
+			var nrr []byte
+			if rm.ae {
+				c.acc.removeServiceImport(rm.from)
+			}
+			if c.pa.reply != nil {
```

Comment:
Here if the client that is exporting the service responds to a request with another request (e.g using `PublishRequest` instead of `Publish`, like `nc.PublishRequest("_INBOX.WWhSTefW", "nowhere", []byte("foo"))`) then the remap is still done even if the client might not be able to handle the response.

Although that type of usage would be rare I think (so we could solve instead for the more general case of discarding unhandled requests), since at this point we know that the client cannot handle the response we could check whether it would be able to handle and skip doing the remap by checking in its sublist by maybe something like:

```diff
diff --git a/server/client.go b/server/client.go
index 1a0f248..6b2bba0 100644
--- a/server/client.go
+++ b/server/client.go
@@ -1762,9 +1762,15 @@ func (c *client) processInboundMsg(msg []byte) {
 				c.acc.removeServiceImport(rm.from)
 			}
 			if c.pa.reply != nil {
-				// We want to remap this to provide anonymity.
-				nrr = c.newServiceReply()
-				rm.acc.addImplicitServiceImport(c.acc, string(nrr), string(c.pa.reply), true)
+				// Check whether the client can handle the published request.
+				rr := c.acc.sl.Match(string(c.pa.reply))
+				canHandleResponse := len(rr.psubs) > 0 || len(rr.qsubs) > 0
+
+				if canHandleResponse {
+					// We want to remap this to provide anonymity.
+					nrr = c.newServiceReply()
+					rm.acc.addImplicitServiceImport(c.acc, string(nrr), string(c.pa.reply), true)
+				}
 			}
```

Reply:
We specifically wanted to allow chaining, or ping ponging as @aricart puts it.